src/index.tsx

```diff
@@ -509,14 +519,18 @@ class App extends React.Component<{}, AppState> {
               parsedElements.length > 0 &&
               parsedElements[0].type // need to implement a better check here...
             ) {
-              clearSelection();
-              parsedElements.forEach(parsedElement => {
+              this.clearSelection();
+
+              parsedElements = parsedElements.map(parsedElement => {
                 parsedElement.x += 10;
                 parsedElement.y += 10;
                 generateDraw(parsedElement);
-                elements.push(parsedElement);
+                return parsedElement;
               });
-              drawScene();
+              this.setState({
+                elements: [...this.state.elements, ...parsedElements]
+              });
+              drawScene(this.state.elements);
```

Comment:
This won't work, setState is asynchronous. You're going to draw the scene before the setState

Reply:
But, if we are redrawing every time the state is updated, we can get rid of all the manual drawScene :)

Reply:
I added the drawScene function call to the setState callback

Reply:
Maybe we should move the drawScene in the componentDidUpdate instead of calling on every setState.